File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int res=0,start=0,end=0,cnt=0;
        while(end<nums.size()){
            if(nums[end]==0) cnt++;
            while(cnt>k){
                if(nums[start]==0) cnt--;
                start++;
            }
            res=max(res,end-start+1);
            end++;
        }
        return res;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        res,start,end,cnt=0,0,0,0
        while(end<len(nums)):
            if(nums[end]==0):
                cnt+=1
            while(cnt>k):
                if(nums[start]==0):
                    cnt-=1
                start+=1
            res=max(res,end-start+1)
            end+=1
        return res